internal/pkg/warewulfd/provision.go

```diff
@@ -193,6 +197,45 @@ func ProvisionSend(w http.ResponseWriter, req *http.Request) {
 			wwlog.ErrorExc(err, "")
 			return
 		}
+	} else if rinfo.stage == "runtime_check" {
```

Comment:
Rather than introduce a new stage, warewulfd should provide ETag or Last-Modified headers, and then support If-None-Match or If-Modified-Since request headers. If it doesn't already, we should also support fetcheing just headers with a HEAD request. This could be implemented just for the runtime stage, but the same approach would also apply to all stages if we wanted it.